### ISSUE TITLE
Debug Failure: tsserver throws after some move refactorings

### ISSUE THREAD

**Issue by cevek** [Microsoft/TypeScript#25458]
**TypeScript Version:**  3.0.0-dev.20180626

**Search Terms:** 

**Code**

```ts
Error: Debug Failure. False expression.
    at mergeSymbol (/usr/local/lib/node_modules/typescript/lib/tsserver.js:29904:26)
    at /usr/local/lib/node_modules/typescript/lib/tsserver.js:29965:49
    at Map.forEach (native)
    at mergeSymbolTable (/usr/local/lib/node_modules/typescript/lib/tsserver.js:29964:20)
    at mergeModuleAugmentation (/usr/local/lib/node_modules/typescript/lib/tsserver.js:29978:17)
    at initializeTypeChecker (/usr/local/lib/node_modules/typescript/lib/tsserver.js:54255:25)
    at Object.createTypeChecker (/usr/local/lib/node_modules/typescript/lib/tsserver.js:29636:9)
    at Object.getTypeChecker (/usr/local/lib/node_modules/typescript/lib/tsserver.js:82503:79)
    at InferredProject.Project.updateGraph (/usr/local/lib/node_modules/typescript/lib/tsserver.js:114332:59)
    at InferredProject.Project.getLanguageService (/usr/local/lib/node_modules/typescript/lib/tsserver.js:114030:26)
    at /usr/local/lib/node_modules/typescript/lib/tsserver.js:118169:51
    at ProjectService.forEachProject (/usr/local/lib/node_modules/typescript/lib/tsserver.js:115200:21)
    at IOSession.Session.getEditsForFileRename (/usr/local/lib/node_modules/typescript/lib/tsserver.js:118161:37)
    at Session.handlers.ts.createMapFromTemplate._a.(anonymous function) (/usr/local/lib/node_modules/typescript/lib/tsserver.js:117018:61)
    at /usr/local/lib/node_modules/typescript/lib/tsserver.js:118343:88
    at IOSession.Session.executeWithRequestId (/usr/local/lib/node_modules/typescript/lib/tsserver.js:118334:28)
    at IOSession.Session.executeCommand (/usr/local/lib/node_modules/typescript/lib/tsserver.js:118343:33)
    at IOSession.Session.onMessage (/usr/local/lib/node_modules/typescript/lib/tsserver.js:118363:35)
    at Interface.<anonymous> (/usr/local/lib/node_modules/typescript/lib/tsserver.js:119618:27)
```

**Comment by mhegazy** [comment]
Can you share your tsserver log file?

**Comment by mhegazy** [comment]
You can upload logs using 
1. With "typescript.tsserver.log": "verbose" enabled
2. Restart vscode or vscode insiders
3. Reproduce the problem
4. Run ```code --upload-logs``` on the command line (or ```code-insiders --upload-logs``` if using vscode insiders) and follow instructions
5. Share the log file ID

**Comment by amcasey** [comment]
The specific refactoring appears to be `getEditsForFileRename`, so @andy-ms might want to take a look.  Also, @sandersn for `mergeSymbolTable`.

**Comment by sandersn** [comment]
@cevek This should be fixed by #25463 in `typescript@next`, which should be available tomorrow.
